cfme/tests/openstack/cloud/test_instances.py

```diff
@@ -178,3 +178,10 @@ def test_delete_instance(new_instance):
         assert False, "entity still exists"
     except ItemNotFound:
         pass
+
+
+def test_instance_operating_system_linux(new_instance):
+    view = navigate_to(new_instance, 'Details')
+    os = view.entities.summary('Properties').get_text_of("Operating System")
+    prov_data_os = new_instance.provider.data['provisioning']['image']['os_distro']
```

Comment:
Since this test is generically marked for openstack providers, it needs to be a lot more lenient in its yaml lookup.

One way to do that is to mark this test for only providers that have this field in the yamls - so that it won't get collected.

Try using ProviderFilter and required_fields - will look _something_ like below, but no promises. :)

Alternatively you can use pytest.skip within the test if the field is empty - using `dict.get()` to safely lookup the yaml value without an AttributeError.

```
@pytest.mark.provider(genfunc=providers, filters=[ProviderFilter(required_fields={'provisioning':{'image':{'os_distro'}}})])
```

Reply:
Done:
used the dict.get() alternative